drizzlepac/alignimages.py

```diff
@@ -450,39 +450,47 @@ def run_align(input_list, archive=False, clobber=False, debug=False, update_hdr_
         # reset process
         temp_imglist = []
         fit_info_dict = OrderedDict()
-        for catalogIndex in range(0, len(
-                catalog_list)):  # loop over astrometric catalog
-            log.info("{} STEP 5: Detect astrometric sources {}".format("-" * 20, "-" * 48))
-            log.info("Astrometric Catalog: %s", str(catalog_list[catalogIndex]))
-            reference_catalog = generate_astrometric_catalog(process_list,
-                                                             catalog=catalog_list[catalogIndex],
-                                                             output=output)
+        reference_catalog_dict={}
+        for algorithm_name in fit_algorithm_list:  # loop over fit algorithm type
+            for catalogIndex in range(0, len(catalog_list)):  # loop over astrometric catalog
+                log.info("{} STEP 5: Detect astrometric sources {}".format("-" * 20, "-" * 48))
+                catalog_name = str(catalog_list[catalogIndex])
+                log.info("Astrometric Catalog: {}".format(catalog_name))
+                # store reference catalogs in a dictionary so that generate_astrometric_catalog() doesn't
+                #  execute unnecessarily after it's been run once for a given astrometric catalog.
+                if catalog_name in reference_catalog_dict.keys():
```

Comment:
Use of `.keys()` here is not functionally necessary, although it is more explicit.

Reply:
Issue resolved by commit [cb068eb](https://github.com/spacetelescope/drizzlepac/pull/328/commits/cb068eb71b8047f72f4f3a8d8951faab996d40c7)